flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/source/enumerator/tracker/SplitTracker.java

```diff
@@ -250,6 +252,24 @@ public Set<String> getKnownSplitIds() {
         return knownSplits.keySet();
     }
 
+    public List<Shard> getKnownShards() {
+        return knownSplits
+                .values()
+                .parallelStream()
+                .filter(
+                        split ->
+                                !ShardUtils.isShardOlderThanInconsistencyDetectionRetentionPeriod(
+                                        split.splitId()))
+                .map(
+                        split ->
+                                Shard.builder()
+                                        .shardId(split.splitId())
+                                        .parentShardId(split.getParentShardId())
+                                        .sequenceNumberRange(SequenceNumberRange.builder().build())
+                                        .build())
```

Comment:
How does this work with the inconsistency tracker? Don't we use the sequence number to detect open/closed shards?

Reply:
yes, you're right, here we are under assumption that there's no inconsistent shard returend from splitTracker so all leaf nodes are open. The shard returned from incremental shard discovery will remove the open leaf node and add itself as a leaf node